### PR TITLE
feat: Refactor for app/db integration and unit tests

### PR DESCRIPTION
**Pull request by FlynnOConnell**
# Adding Tests :tada:

- [x] Refactor main.rs: split into functions that return a db `pool` and axum `app` instance to be used in `main` and `tests`.
- [x] Spin up a separate instance of the app and database that is cleaned on each test
- [ ] Run these tests in a workflow utilizing `sqlx::test` and offline mode
- [ ] Create workflow to automate database and app instances

## Notes

- Do we want each query to be separate, or sequential i.e. create and login user -> create message -> channel all in 1 test, or separate
- The database needs to be setup via docker so that it can be run via a workflow

## Links
- [CI to load database schema and sample data, 3x per day](https://stackoverflow.com/a/145163)

### CODE REVIEW

File: backend/src/configuration/mod.rs [hunk 1]
```diff
@@ -4,19 +4,16 @@ use std::env;
...
 
-/// Fetches the port number from env var PORT or None
 fn load_port() -> Option<FlukePort> {
     let env_port: String = env::var("PORT").ok()?;
     let port = env_port.parse::<u16>().ok()?;
     Some(FlukePort::from(port))
 }
 
-/// Loads from a .env file the configuration for Fluke
 pub fn load_config() -> FlukeConfiguration {
```
Comment (LukeBrandon) [inline backend/src/configuration/mod.rs:17]: why did you remove these

File: backend/src/configuration/models.rs [hunk 2]
```diff
@@ -15,9 +15,7 @@ impl From<u16> for FlukePort {
 }
 
 pub struct FlukeConfiguration {
-    /// REQUIRED! There is not default value
     pub database_url: String,
-    /// Default 8000
```
Comment (LukeBrandon) [inline backend/src/configuration/models.rs]: why did you remove these

File: backend/src/configuration/mod.rs (hunk 1 above)
Comment (FlynnOConnell) [reply backend/src/configuration/mod.rs:17]: Described what the code was doing when the code itself did a good job of that

File: backend/src/configuration/models.rs (hunk 2 above)
Comment (FlynnOConnell) [reply backend/src/configuration/models.rs]: Described what the code was doing when the code itself did a good job of that
